core/templates/components/common-layout-directives/navigation-bars/top-navigation-bar.component.spec.ts

```diff
@@ -554,6 +603,61 @@ describe('TopNavigationBarComponent', () => {
           AppConstants.DEFAULT_PROFILE_IMAGE_WEBP_PATH));
     }));
 
+  it('should fetch the number of unread feedback' +
+  'when user is logged In', fakeAsync(() => {
+    let userInfo = new UserInfo(
+      ['USER_ROLE'], true, false, false, false, true,
+      'en', 'username1', 'tester@example.com', true
+    );
+
+    spyOn(component, 'truncateNavbar').and.stub();
+    spyOn(userService, 'getUserInfoAsync').and.resolveTo(userInfo);
+    const fetchDataSpy = spyOn(
+      feedbackUpdatesBackendApiService,
+      'fetchFeedbackUpdatesDataAsync').and.returnValue(Promise.resolve({
+      numberOfUnreadThreads: FeedbackUpdatesData.
+        number_of_unread_threads,
+      threadSummaries: (
+        FeedbackUpdatesData.thread_summaries.map(
+          threadSummary => FeedbackThreadSummary
+            .createFromBackendDict(threadSummary))),
+      paginatedThreadsList: []
+    }));
+
+    component.userIsLoggedIn = true;
+    component.ngOnInit();
+    tick();
+
+    expect(component.unreadThreadsCount).toBe(10);
+    expect(fetchDataSpy).toHaveBeenCalled();
+  }));
+
+  it('should show an alert when fails to' +
+  'get the feedback updates data', fakeAsync(() => {
+    let userInfo = new UserInfo(
+      ['USER_ROLE'], true, false, false, false, true,
+      'en', 'username1', 'tester@example.com', true
+    );
+
+    spyOn(component, 'truncateNavbar').and.stub();
+    spyOn(userService, 'getUserInfoAsync').and.resolveTo(userInfo);
+    const fetchDataSpy = spyOn(
+      feedbackUpdatesBackendApiService,
+      'fetchFeedbackUpdatesDataAsync')
+      .and.rejectWith(404);
+    const alertsSpy = spyOn(alertsService, 'addWarning').and.callThrough();
+
+    component.userIsLoggedIn = true;
+    component.ngOnInit();
+    tick(1010);
+    fixture.detectChanges();
+
+    expect(alertsSpy).toHaveBeenCalledWith(
+      'Failed to get number of unread thread of feedback updates');
+    expect(fetchDataSpy).toHaveBeenCalled();
+    tick(1010);
```

Comment:
Why is tick present here?

Reply:
To remove 2 timer error.
![image](https://github.com/oppia/oppia/assets/100484672/37ac2218-a0cd-4098-a559-1023c68d2d92)

Reply:
You will have to use `flush();`